src/App.js

```diff
@@ -1,18 +1,26 @@
 import React from 'react';
 import ToDoList from './ToDoList';
 import AddToDoForm from './AddTodoForm';
-
+function useSemiPersistentState() {
+  const [todoList, setTodoList] = React.useState(
+    JSON.parse(localStorage.getItem("savedTodoList")) || []
+    );
+  React.useEffect(() => {
+    localStorage.setItem("savedTodoList", JSON.stringify(todoList));
+  }, [todoList]);
+  return [todoList,setTodoList];
+}
 function App() {
-  const [todoList, setTodoList] = React.useState([]);
+  const [todoList,setTodoList] = useSemiPersistentState('savedTodoList', []);
```

Comment:
The `useSemiPersistentState` hook is called with two arguments, 'savedTodoList' and an empty array. However, the hook definition does not accept any arguments. This would cause an error. The hook should be called without arguments.